Chatkit/Redux/Store/Store.swift

```diff
@@ -1,21 +1,22 @@
-
-protocol StoreDelegate: AnyObject {
-    func store(_ store: Store, didUpdateState state: MasterState)
-}
-
-protocol HasStore {
-    var store: Store { get }
-}
+import Foundation
 
 protocol Store {
-    var state: MasterState { get }
+    
+    var state: VersionedState { get }
```

Comment:
It was a deliberate choice of my to *not* expose the `state` of the `Store` directly.  Things that are interested in the Store should be always be registering as a Listener.  I don't believe there's any requirement to expose this because I've removed it in your branch and the code & tests still compile and pass.

Reply:
If you are happy to remove it then the `private(set) var state: VersionedState` on the `ConcreteStore` can change to `private var state: VersionedState` also.

Reply:
I can delete `state` from the protocol. However, it feels a bit weird to me (and potentially error prone) that the only way to read the current state of the store would be to maintain a local copy of that state in a `StoreListner`. I was under the impression that `Store` is supposed to be the single source of truth for `State`.

Reply:
This change will also break quite a number of existing unit tests for `ConcreteStore`.

Reply:
So as discussed between us and Mike, the conclusion was that we should..
a) Not expose `state` on the protocol
b) but leave `state` on `ConcreteStore` as `internal` so that the unit tests are able to inspect its value.

Reply:
I've deleted the property from the protocol.